### ISSUE TITLE
[Bug]: Retry the failed index build task:fail reason=NoSuchKey after mixcoord recovered from oomkilled

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#30580]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: cardinal-milvus-io-2.3-b2d3278-20240206
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka): pulsar   
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. collection `laion_cpu_1` has 50m-768d data, HNSW index
2. load collection with 2 replicas
3. concurrent requests: insert + delete + flush + search + query from 2024-02-06 16:00 to 2024-02-07 02:14:01
```
'concurrent_params': {'concurrent_number': 100,
                                   'during_time': '10h',
                                   'interval': 120,
                                   'spawn_rate': None},
             'concurrent_tasks': [{'type': 'query',
                                   'weight': 2,
                                   'params': {'expr': '50000000 '
                                                      '< '
                                                      'id '
                                                      '< '
                                                      '5010000',
                                              'timeout': 1200}},
                                  {'type': 'search',
                                   'weight': 7,
                                   'params': {'nq': 10,
                                              'top_k': 100,
                                              'random_data': True,
                                              'search_param': {'ef': 100},
                                              'timeout': 600}},
                                  {'type': 'insert',
                                   'weight': 7,
                                   'params': {'nb': 200,
                                              'start_id': 50000000,
                                              'random_id': True,
                                              'random_vector': True,
                                              'timeout': 600}},
                                  {'type': 'delete',
                                   'weight': 4,
                                   'params': {'delete_length': 100,
                                              'timeout': 600}},
                                  {'type': 'flush',
                                   'weight': 1,
                                   'params': {'timeout': 600}}]}
```
4. mixcoord oomkilled and 134 error code from 2024-02-06 17:37:45 ([pod_container_status_last_terminated_reason of mc-qqg4d](https://grafana-4am.zilliz.cc/explore?panes=%7B%22nhf%22:%7B%22datasource%22:%22P1809F7CD0C75ACF3%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22kube_pod_container_status_last_terminated_reason%7Bpod%3D%5C%22laion1b-test-2-milvus-mixcoord-868c566c7c-qqg4d%5C%22%7D%22,%22range%22:true,%22instant%22:true,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22P1809F7CD0C75ACF3%22%7D,%22editorMode%22:%22code%22,%22legendFormat%22:%22__auto%22%7D%5D,%22range%22:%7B%22from%22:%221707238554225%22,%22to%22:%221707288522225%22%7D%7D%7D&schemaVersion=1&orgId=1)
![image](https://github.com/milvus-io/milvus/assets/27288593/2768f11b-f40d-40b3-9ef1-11e9bf79ee8f)
```
630 panic: runtime error: invalid memory address or nil pointer dereference
   631 [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3caf08e]
   632 
   633 goroutine 9377 [running]:
   634 panic({0x4b333a0, 0x74ac860})
   635     /usr/local/go/src/runtime/panic.go:987 +0x3bb fp=0xc00599bcb8 sp=0xc00599bbf8 pc=0x1a99cdb
   636 runtime.panicmem(...)
   637     /usr/local/go/src/runtime/panic.go:260
   638 runtime.sigpanic()
   639     /usr/local/go/src/runtime/signal_unix.go:841 +0x37d fp=0xc00599bd18 sp=0xc00599bcb8 pc=0x1ab1fdd
   640 github.com/milvus-io/milvus/internal/distributed/datanode/client.wrapGrpcCall[...]({0x56e7a40, 0xc16649d590?}, 0x0, 0xc05c0d50a0)
   641     /go/src/github.com/milvus-io/milvus/internal/distributed/datanode/client/client.go:90 +0xae fp=0xc00599bd50 sp=0xc00599bd18 pc=0x       3caf08e
   642 github.com/milvus-io/milvus/internal/distributed/datanode/client.(*Client).GetCompactionState(0xc00599be90?, {0x56e7a40?, 0xc16649d59       0}, 0x0?, {0x0?, 0x1a72b9d?, 0x1?})
```

4. mixcoord recovered from terminated at 2024-02-06 18:34:08(UTC), dataNode start to do compaction task, but index task failed always
![image](https://github.com/milvus-io/milvus/assets/27288593/5b825bd3-435f-4d63-afcf-d2871e00a0ca)
 
- [index retry and NoSuchKey logs](https://grafana-4am.zilliz.cc/explore?panes=%7B%22Qa9%22:%7B%22datasource%22:%22vhI6Vw67k%22,%22queries%22:%5B%7B%22refId%22:%22C%22,%22expr%22:%22%7Bcluster%3D%5C%224am%5C%22,%20namespace%3D%5C%22qa-milvus%5C%22,%20pod%3D~%5C%22laion1b-test-2-milvus-mixcoord-868c566c7c-qqg4d%5C%22%7D%20%7C~%20%5C%22447543888699353988%5C%22%20%7C~%20%5C%22segment%5C%22%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22vhI6Vw67k%22%7D,%22editorMode%22:%22code%22,%22hide%22:true%7D,%7B%22refId%22:%22A%22,%22expr%22:%22%20%7Bcluster%3D%5C%224am%5C%22,%20namespace%3D%5C%22qa-milvus%5C%22,%20pod%3D~%5C%22laion1b-test-2-milvus-datanode-5989b844f5-zm6lp%5C%22%7D%20%7C~%20%5C%22UpdateChannelCheckpoint%20success%7CgetChannelCheckpoint%20for%20segment%5C%22%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22vhI6Vw67k%22%7D,%22editorMode%22:%22code%22,%22hide%22:true%7D,%7B%22refId%22:%22B%22,%22expr%22:%22%7Bcluster%3D%5C%224am%5C%22,%20namespace%3D%5C%22qa-milvus%5C%22,%20pod%3D~%5C%22laion1b-test-2-milvus-mixcoord-868c566c7c-qqg4d%5C%22%7D%20%20%7C~%20%5C%22index%5C%22%20%7C~%20%5C%22reason%5C%22%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22vhI6Vw67k%22%7D,%22editorMode%22:%22code%22%7D%5D,%22range%22:%7B%22from%22:%22now-2d%22,%22to%22:%22now%22%7D%7D%7D&schemaVersion=1&orgId=1)
![image](https://github.com/milvus-io/milvus/assets/27288593/33e82d63-d03a-450a-9714-fe3a86ccad5d)

- some warn logs [mixcoord warn logs](https://grafana-4am.zilliz.cc/explore?panes=%7B%22Qa9%22:%7B%22datasource%22:%22vhI6Vw67k%22,%22queries%22:%5B%7B%22refId%22:%22C%22,%22expr%22:%22%7Bcluster%3D%5C%224am%5C%22,%20namespace%3D%5C%22qa-milvus%5C%22,%20pod%3D~%5C%22laion1b-test-2-milvus-mixcoord-868c566c7c-qqg4d%5C%22%7D%20%7C%3D%20%5C%22WARN%5C%22%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22vhI6Vw67k%22%7D,%22editorMode%22:%22code%22,%22hide%22:false%7D%5D,%22range%22:%7B%22from%22:%221707244448000%22,%22to%22:%221707292800000%22%7D%7D%7D&schemaVersion=1&orgId=1)
![image](https://github.com/milvus-io/milvus/assets/27288593/6d1ff888-5506-483d-9d72-b331a5897f25)

### Steps To Reproduce

```markdown
argo: https://argo-workflows.zilliz.cc/archived-workflows/qa/38f09676-d436-42f4-8e43-2840f8931214?nodeId=laion1b-test-cron-1707235200
```

### Milvus Log

pods:
```
laion1b-test-2-etcd-0                                             1/1     Running            1 (26d ago)      54d     10.104.25.31    4am-node30   <none>           <none>
laion1b-test-2-etcd-1                                             1/1     Running            0                54d     10.104.30.94    4am-node38   <none>           <none>
laion1b-test-2-etcd-2                                             1/1     Running            0                54d     10.104.34.225   4am-node37   <none>           <none>
laion1b-test-2-milvus-datanode-5989b844f5-pw5sg                   1/1     Running            4 (9h ago)       24h     10.104.13.187   4am-node16   <none>           <none>
laion1b-test-2-milvus-datanode-5989b844f5-zm6lp                   1/1     Running            2 (3m24s ago)    24h     10.104.1.111    4am-node10   <none>           <none>
laion1b-test-2-milvus-indexnode-7bb59785b5-46clt                  1/1     Running            0                24h     10.104.13.188   4am-node16   <none>           <none>
laion1b-test-2-milvus-indexnode-7bb59785b5-7hfxj                  1/1     Running            0                24h     10.104.5.80     4am-node12   <none>           <none>
laion1b-test-2-milvus-indexnode-7bb59785b5-gfp8l                  1/1     Running            0                24h     10.104.9.164    4am-node14   <none>           <none>
laion1b-test-2-milvus-indexnode-7bb59785b5-jlrnw                  1/1     Running            0                24h     10.104.14.85    4am-node18   <none>           <none>
laion1b-test-2-milvus-indexnode-7bb59785b5-knxwn                  1/1     Running            0                24h     10.104.33.102   4am-node36   <none>           <none>
laion1b-test-2-milvus-mixcoord-868c566c7c-qqg4d                   1/1     Running            15 (13h ago)     24h     10.104.18.222   4am-node25   <none>           <none>
laion1b-test-2-milvus-proxy-64b6d7787-zck8d                       1/1     Running            1 (24h ago)      24h     10.104.12.248   4am-node17   <none>           <none>
laion1b-test-2-milvus-querynode-1-6f8889c79b-2j46t                1/1     Running            0                24h     10.104.31.49    4am-node34   <none>           <none>
laion1b-test-2-milvus-querynode-1-6f8889c79b-jfbtf                1/1     Running            0                24h     10.104.17.179   4am-node23   <none>           <none>
laion1b-test-2-milvus-querynode-1-6f8889c79b-nv75z                1/1     Running            0                23h     10.104.32.197   4am-node39   <none>           <none>
laion1b-test-2-milvus-querynode-1-6f8889c79b-w5dg2                1/1     Running            0                24h     10.104.28.110   4am-node33   <none>           <none>
laion1b-test-2-pulsar-bookie-0                                    1/1     Running            0                54d     10.104.33.107   4am-node36   <none>           <none>
laion1b-test-2-pulsar-bookie-1                                    1/1     Running            0                30d     10.104.18.240   4am-node25   <none>           <none>
laion1b-test-2-pulsar-bookie-2                                    1/1     Running            0                54d     10.104.25.32    4am-node30   <none>           <none>
laion1b-test-2-pulsar-broker-0                                    1/1     Running            0                48d     10.104.1.69     4am-node10   <none>           <none>
laion1b-test-2-pulsar-proxy-0                                     1/1     Running            0                54d     10.104.4.218    4am-node11   <none>           <none>
laion1b-test-2-pulsar-recovery-0                                  1/1     Running            0                54d     10.104.14.151   4am-node18   <none>           <none>
laion1b-test-2-pulsar-zookeeper-0                                 1/1     Running            0                54d     10.104.29.87    4am-node35   <none>           <none>
laion1b-test-2-pulsar-zookeeper-1                                 1/1     Running            0                54d     10.104.21.124   4am-node24   <none>           <none>
laion1b-test-2-pulsar-zookeeper-2                                 1/1     Running            0                54d     10.104.34.229   4am-node37   <none>           <none>
```

### Anything else?
grafana metrics: https://grafana-4am.zilliz.cc/d/uLf5cJ3Ga/milvus2-0?orgId=1&from=1707233864017&to=1707293652817&var-datasource=P1809F7CD0C75ACF3&var-cluster=&var-namespace=qa-milvus&var-instance=laion1b-test-2&var-collection=All&var-app_name=milvus

_No response_

**Comment by xiaocai2333** [comment]
![image](https://github.com/milvus-io/milvus/assets/46207236/9898a36c-d828-4c16-92d3-171185a68907)
![image](https://github.com/milvus-io/milvus/assets/46207236/ecb01f61-cebe-4563-bf78-89f549e61760)

DataCoord did not obtain compaction results from the datanode for nearly 6 hours, causing the files generated by compaction to be GC.

**Comment by xiaofan-luan** [comment]
There might be two things we need to figure out:
1. why it stucked for 6 years, I guess the request is queued somewhere (Did we check etcd)
2. we might need a check here, when save bin log path, we need to check if the file is generated for too long ago(like one hour)

**Comment by ThreadDao** [comment]
Update: after a few hours (without any new requests) the two dataNodes panic many times.
[dn_zm6lp_pre.log](https://github.com/milvus-io/milvus/files/14196502/dn_zm6lp_pre.log)
[dn_pw5sg_pre.log](https://github.com/milvus-io/milvus/files/14196505/dn_pw5sg_pre.log)

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x389644c]

goroutine 143273 [running]:
panic({0x4b333a0, 0x74ac860})
    /usr/local/go/src/runtime/panic.go:987 +0x3bb fp=0xc004fd06a8 sp=0xc004fd05e8 pc=0x1a99cdb
runtime.panicmem(...)
    /usr/local/go/src/runtime/panic.go:260
runtime.sigpanic()
    /usr/local/go/src/runtime/signal_unix.go:841 +0x37d fp=0xc004fd0708 sp=0xc004fd06a8 pc=0x1ab1fdd
github.com/milvus-io/milvus/internal/storage.(*InsertBinlogIterator).Next(0xc001e8ec60)
    /go/src/github.com/milvus-io/milvus/internal/storage/binlog_iterator.go:95 +0x18c fp=0xc004fd07d0 sp=0xc004fd0708 pc=0x389644c
github.com/milvus-io/milvus/internal/datanode.(*compactionTask).merge(0xc0021ff600, {0x56e7a40, 0xc003b1b0b0}, {0xc0012f5200, 0x21, 0x150?}, 0xc0015e5348?, 0x1a6ca0a?, 0xc001511900, 0xc0022b0570)
    /go/src/github.com/milvus-io/milvus/internal/datanode/compactor.go:335 +0x1b4d fp=0xc004fd12a8 sp=0xc004fd07d0 pc=0x401074d
github.com/milvus-io/milvus/internal/datanode.(*compactionTask).compact(0xc0021ff600)
    /go/src/github.com/milvus-io/milvus/internal/datanode/compactor.go:637 +0x11bf fp=0xc004fd1d60 sp=0xc004fd12a8 pc=0x40124df
github.com/milvus-io/milvus/internal/datanode.(*compactionExecutor).executeTask(0xc0008c9a40, {0x56fa130, 0xc0021ff600})
    /go/src/github.com/milvus-io/milvus/internal/datanode/compaction_executor.go:96 +0x315 fp=0xc004fd1fb8 sp=0xc004fd1d60 pc=0x400d3f5
github.com/milvus-io/milvus/internal/datanode.(*compactionExecutor).executeWithState.func1()
    /go/src/github.com/milvus-io/milvus/internal/datanode/compaction_executor.go:75 +0x2e fp=0xc004fd1fe0 sp=0xc004fd1fb8 pc=0x400cfae
runtime.goexit()
    /usr/local/go/src/runtime/asm_amd64.s:1598 +0x1 fp=0xc004fd1fe8 sp=0xc004fd1fe0 pc=0x1ad3b21
created by github.com/milvus-io/milvus/internal/datanode.(*compactionExecutor).executeWithState
    /go/src/github.com/milvus-io/milvus/internal/datanode/compaction_executor.go:75 +0x95
```
- datanode pods
```
laion1b-test-2-milvus-datanode-5989b844f5-pw5sg                   1/1     Running            24 (2m18s ago)   32h
laion1b-test-2-milvus-datanode-5989b844f5-zm6lp                   1/1     Running            10 (113m ago)    32h
```

[metrics of laion1b-test-2](https://grafana-4am.zilliz.cc/d/uLf5cJ3Ga/milvus2-0?orgId=1&from=1707290162771&to=1707323158047&var-datasource=P1809F7CD0C75ACF3&var-cluster=&var-namespace=qa-milvus&var-instance=laion1b-test-2&var-collection=All&var-app_name=milvus)

**Comment by xiaofan-luan** [comment]
This is gonna to be the first priority to solve

**Comment by xiaocai2333** [comment]
The location of code panic:
// merge binlog of compaction
![image](https://github.com/milvus-io/milvus/assets/46207236/8ac8fc1b-2aa9-4356-989a-b081b7739b3e)

Get the binlog from log:
```
[2024/02/07 14:30:34.261 +00:00] [DEBUG] [datanode/binlog_io.go:73] ["down load"] [path="[
files/insert_log/447180040934991941/447180040934992003/447534060537469568/0/447534060536192932,
files/insert_log/447180040934991941/447180040934992003/447534060537469568/102/447534060536192922,
files/insert_log/447180040934991941/447180040934992003/447534060537469568/106/447534060536192926,
files/insert_log/447180040934991941/447180040934992003/447534060537469568/107/447534060536192927,
files/insert_log/447180040934991941/447180040934992003/447534060537469568/108/447534060536192928,
files/insert_log/447180040934991941/447180040934992003/447534060537469568/111/447534060536192931]"]
```
It can be seen from log the number of binlog is wrong, and there are a few fields missing.
The binlog of the segment that can be loaded normally:
```
[2024/02/07 14:30:34.067 +00:00] [DEBUG] [datanode/binlog_io.go:73] ["down load"] [path="[
files/insert_log/447180040934991941/447180040934992003/447543888698249596/100/447543888697848068,
files/insert_log/447180040934991941/447180040934992003/447543888698249596/102/447543888697848070,
files/insert_log/447180040934991941/447180040934992003/447543888698249596/103/447543888697848071,
files/insert_log/447180040934991941/447180040934992003/447543888698249596/104/447543888697848072,
files/insert_log/447180040934991941/447180040934992003/447543888698249596/105/447543888697848073,
files/insert_log/447180040934991941/447180040934992003/447543888698249596/107/447543888697848075,
files/insert_log/447180040934991941/447180040934992003/447543888698249596/106/447543888697848074,
files/insert_log/447180040934991941/447180040934992003/447543888698249596/109/447543888697848077,
files/insert_log/447180040934991941/447180040934992003/447543888698249596/1/447543888697848081,
files/insert_log/447180040934991941/447180040934992003/447543888698249596/101/447543888697848069,
files/insert_log/447180040934991941/447180040934992003/447543888698249596/108/447543888697848076,
files/insert_log/447180040934991941/447180040934992003/447543888698249596/110/447543888697848078,
files/insert_log/447180040934991941/447180040934992003/447543888698249596/111/447543888697848079,
files/insert_log/447180040934991941/447180040934992003/447543888698249596/0/447543888697848080]"]
```

However, we still don't know why the binlog is missing, the previously saved log is not complete enough.

**Comment by xiaocai2333** [comment]
> ![image](https://private-user-images.githubusercontent.com/46207236/302909222-9898a36c-d828-4c16-92d3-171185a68907.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg0OTcxNjYsIm5iZiI6MTcwODQ5Njg2NiwicGF0aCI6Ii80NjIwNzIzNi8zMDI5MDkyMjItOTg5OGEzNmMtZDgyOC00YzE2LTkyZDMtMTcxMTg1YTY4OTA3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIxVDA2Mjc0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNlNTYyYmYyMjAyZGY3MmQ0ODUzY2NmMjZiZDI4NDJiZjI2OTVkOTlmNjQ5OGY4MjQ1YjhlNGNmODg1MWZkZTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.n3aoLvy8Mra2nrySFg33ro4vZRigC5OF-N4zTuHHGGo) ![image](https://private-user-images.githubusercontent.com/46207236/302909385-ecb01f61-cebe-4563-bf78-89f549e61760.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg0OTcxNjYsIm5iZiI6MTcwODQ5Njg2NiwicGF0aCI6Ii80NjIwNzIzNi8zMDI5MDkzODUtZWNiMDFmNjEtY2ViZS00NTYzLWJmNzgtODlmNTQ5ZTYxNzYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIxVDA2Mjc0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4ZTMzZmEwNzQ1N2JmNjMyZTJmMzkxNzNhMWQ2OTdkMDBjM2Q2NDQ2MjBjMDQ3NjBmZThjZjFmYmM5NzI1MzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xS1M8wz1hLtYKiA6ba-nIjajJxtYwaU-iwFscZ6wBJs)
> 
> DataCoord did not obtain compaction results from the datanode for nearly 6 hours, causing the files generated by compaction to be GC.

Even if the binlog is mistakenly gc'd for this reason, shouldn't it exist in the meta?

**Comment by xiaofan-luan** [comment]
could this be related to compress of log path?

due to some reason, logs don't have same segmentid in the same segment, which cause the following problem

**Comment by xiaocai2333** [comment]
It seems that found the datacoord why has not obtained compaction result from datanode for about six hours.

Look the log:
![image](https://github.com/milvus-io/milvus/assets/46207236/c8d19929-3454-4bf3-84f6-d3a6710639b0)
and the code:
![image](https://github.com/milvus-io/milvus/assets/46207236/408e797b-3e46-4aa4-952d-386387b91372)
It can be seen that the above two logs should be consistent, but they were blocked for nearly 11 hours.
Analyzing the code, we can see that the only place that can be blocked here is that flushCh is full, as shown in the green box in the picture.
check the log:
![image](https://github.com/milvus-io/milvus/assets/46207236/b217592a-c5d4-4a5d-b5e8-07f82b06976f)

found the segment: 447859087914022887 is blocked a few hours in `postFlush`.
check the code:
![image](https://github.com/milvus-io/milvus/assets/46207236/c6ad5e24-1451-49d1-babe-e3da9d090b45)

found another channel that may be blocked.
check the code:
![image](https://github.com/milvus-io/milvus/assets/46207236/ebd66149-5da2-4792-8c0c-ebd6f13fbd10)

This channel may be blocked by a timer. Because there are many segments and the timer interval is 1 minute, there is a high possibility that after the timer trigger task is executed, the next timer has arrived.

solution：
Try to stuff segmentID into  `idIndexCh` channel, skip if the queue is full.

**Comment by ThreadDao** [comment]
Fixed `cardinal-milvus-io-2.3-ef086dc-20240222`
